ios/FluentUI.Demo/FluentUI.Demo/Demos/ColorDemoController.swift

```diff
@@ -147,19 +146,23 @@ class ColorDemoController: UIViewController {
 
     @objc private func segmentedControlValueChanged(sender: Any) {
         if let segmentedControl = sender as? SegmentedControl {
-            let windowType = colorProviderThemedWindowTypes[segmentedControl.selectedSegmentIndex].windowType
-            let colorThemeHost = view.window?.windowScene?.delegate as? ColorThemeHosting
-
-            if let navigationController = navigationController {
-                navigationController.popViewController(animated: false)
-                colorThemeHost?.updateToWindowWith(type: windowType, pushing: self)
+            let window = colorProviderThemedWindows[segmentedControl.selectedSegmentIndex].window
+            if let colorProvider = window as? ColorProviding {
+                Colors.setProvider(provider: colorProvider, for: window)
             }
         }
+
+        tableView.reloadData()
     }
 
+    private let tableView = UITableView(frame: .zero, style: .grouped)
     private let colorProviderThemedWindowTypes: [(name: String, windowType: UIWindow.Type)] = [("Default", DemoColorThemeDefaultWindow.self),
```

Comment:
instead of 2 different dictionary, could we make dictionary with string and a new struct, or just array of custom object to make sure these are aligned all the time?